Assigment 2

RodrigoDuranAndres opened the pull request.

dgarijo commented on Assignment4/Rodrigo-c20c033/task06.py:
```diff
@@ -0,0 +1,75 @@
+**Task 06: Modifying RDF(s)**
+"""
+
+!pip install rdflib 
```
This will not work in Python. You need to comment out this line (the same applies for other tasks)

dgarijo commented on Assignment4/Rodrigo-c20c033/task06.py:
```diff
@@ -0,0 +1,75 @@
+
+# TO DO
+g.add((ns.UPM, RDF.type, ns.University))
+g.add((ns.Researcher, VCARD.Work, ns.University))
```
This is wrong.  You cannot add properties between classes. Properties define the data model. You may state their domain and range with classes, but not like this

dgarijo commented on Assignment4/Rodrigo-c20c033/task07.py:
```diff
@@ -0,0 +1,122 @@
+  q3 = prepareQuery('''
+    SELECT ?Instance ?Subclases WHERE { 
+      OPTIONAL{?Instance RDF:type '''+ f"<{clase}>"+'''}
+      OPTIONAL{?Subclases RDFS:subClassOf '''+ f"<{clase}>"+'''}
```
While this may work, it is not the correct way of doing it.
You can add a statement directly in sparql to retrieve only subclasses of Person, and use that in a very similar manner you do here. Just not precalculating the classes first, which defeats the purpose.
something like: select ?i where{ ?i rdf:type ?class. ?class rdfs:subclassOf* ex:Person}

dgarijo commented on Assignment4/Rodrigo-c20c033/task07.py:
```diff
@@ -0,0 +1,122 @@
+          OPTIONAL{'''+ f"<{r.Instance}>"+''' ?P ?O}
+        }
+        ''',
+        initNs = {"RDFS": RDFS, "RDF":RDF}
```
You are separating this in 2 queries. It should be done with only one query

RodrigoDuranAndres replied:
done

dgarijo reviewed (changes requested):
You need to improve assignment 4. Also, you are removing the original notebooks! I cannot merge until you undo that

ocorcho commented:
Given all the errors identified, we will close this pull request, and you will have the opportunity to resubmit later all of the assignments, one by one, if you wish.

RodrigoDuranAndres commented:
Good morning.

I changed my push and I added an answer to your request saying I have 
changed it. I was eliminating two notebooks that I shouldn't (dum error) 
and I changed that, and also I changed all the assignments that you 
write a comment about.
Did I make more mistakes?
What can I do next time to make you notice changes I made in the push?

Sorry, Rodrigo.

El 2021-12-14 10:27, Oscar Corcho escribió:
> Given all the errors identified, we will close this pull request, and
> you will have the opportunity to resubmit later all of the
> assignments, one by one, if you wish.
> 
> --
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub [1], or unsubscribe
> [2].
> Triage notifications on the go with GitHub Mobile for iOS [3] or
> Android [4].
> 
> Links:
> ------
> [1] 
> https://github.com/FacultadInformatica-LinkedData/Curso2021-2022-DataScience/pull/28#issuecomment-993343297
> [2] 
> https://github.com/notifications/unsubscribe-auth/AVXKA742KHIF67RNZMC4TTTUQ4EZZANCNFSM5FL6AQYA
> [3] 
> https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&amp;mt=8&amp;pt=524675
> [4] 
> https://play.google.com/store/apps/details?id=com.github.android&amp;referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub

ocorcho commented:
Do a fresh pull request now. Before submitting, make sure that you fetch from the global repository, so as to make sure that you have the latest version and there are no errors.
